adventurelib.py

```diff
@@ -326,55 +338,119 @@ def no_command_matches(command):
 def when(command, **kwargs):
     """Decorator for command functions."""
     def dec(func):
+        """decorator"""
         _register(command, func, kwargs)
         return func
     return dec
 
 
-def help():
+def cmd_help():
     """Print a list of the commands you can give."""
     print('Here is a list of the commands you can give:')
     cmds = sorted(c.orig_pattern for c, _, _ in commands)
     for c in cmds:
         print(c)
 
 
-def _handle_command(cmd):
+def _handle_command(cmd, game=None):
     """Handle a command typed by the user."""
     ws = cmd.lower().split()
     for pattern, func, kwargs in commands:
         args = kwargs.copy()
         matches = pattern.match(ws)
         if matches is not None:
             args.update(matches)
-            func(**args)
+            func(**args, game=game)
             break
     else:
         no_command_matches(cmd)
     print()
 
 
-def start(help=True):
+class Interface:
+    """Superclass for all interfaces (ways of interacting with the game)"""
+    def __init__(self):
+        pass
+    
+    def get_command(self, prompt):
+        """Get a command"""
+        return ''
+    
+    def say(self, msg):
+        """Send output to the user"""
+        pass
+
+        
+class TerminalInterface(Interface):
```

Comment:
Ah, that's a good idea, but I'm not certain how to do that. To be clear, the reason I have it set up this way is so that other interfaces- such as an interface to a webpage or a GUI- could be made. So just always delegating to TerminalInterface would break things.